### ISSUE TITLE
Backticks in field descriptions

### ISSUE THREAD

**Issue by hwlmatt** [wp-graphql/wp-graphql#514]
**The Question / Request** 
What is the intended purpose of using backticks inside the description fields? 
If it is just to show emphasis or quotations, since the single quote already creates a literal, double quotes or no quotes would be fine, yes? 

Or are the backticks creating a literal inside the single quote literal for some other purpose I am unaware of?

Perhaps a style guide determination for the preferred pattern:
```
'description' => __( 'Some description text with quotes around a \'$variable\' here.', 'wp-graphql' ),
'description' => __( 'Some description text with quotes around a "$variable" here.', 'wp-graphql' ),
'description' => __( 'Some description text with quotes around a `$variable` here.', 'wp-graphql' ),
'description' => __( 'Some description text with quotes around a $variable here.', 'wp-graphql' ),

```
Conforming Style for all fields:
In consideration of a user/plugin extending definitions, perhaps a sanitization method to conform all values would be appropriate.

More Hooks:
If a hook does not exist, I would propose a hook similar to [`graphql_field_definition`](https://github.com/wp-graphql/wp-graphql/blob/develop/src/Utils/InstrumentSchema.php#L77)
as part of the return method for any other fields.
For instance before the `return` in the [`fields()` method of TermObjectConnectionArgs.php](https://github.com/wp-graphql/wp-graphql/blob/develop/src/Type/TermObject/Connection/TermObjectConnectionArgs.php#L56).

**The workflow Issue backticks in field descriptions caused for me**

I found several (35?) instances where the description fields have variables encased in backticks, rather than escaped single quotes, double quotes, or no quotes at all. 
For instance: 
WITH:
```
"""
  Retrieve terms where the description is LIKE `$description__like`. Default empty.
  """
  descriptionLike: String
```
WITHOUT:
```
"""
  Array of term ids to exclude along with all of their descendant terms. If
  $include is non-empty, $exclude_tree is ignored. Default empty array.
  """
  excludeTree: [Int]
```

This was causing a work flow issue in converting schema to javascript const, as it involves encasing the entire string as backticks.

For instance with Apollo Server Express' GQL:
```
import { gql } from 'apollo-server-express';
const wpWholeSchema = gql`
schema {
  query: RootQuery
  mutation: RootMutation
}

...
input ChildrenCategoriesTermArgs {
...
... **inherited backtick below closes string**
  """
  Retrieve terms where the description is LIKE `$description__like`. Default empty. 
  """
  descriptionLike: String
...
`; 
```

I was unable to find all the specific filter hooks to edit them out, but the [`graphql_field_definition`](https://github.com/wp-graphql/wp-graphql/blob/develop/src/Utils/InstrumentSchema.php#L77) hook gave me access to maybe half of them, but I did not find a hook that gives access to those fields defined in [TermObjectConnectionArgs.php](https://github.com/wp-graphql/wp-graphql/blob/develop/src/Type/TermObject/Connection/TermObjectConnectionArgs.php#L56)

For now, I'm just filtering the entire text-domain of the plugin with WP's `gettext` hook like so:
```
add_filter( 'gettext', 'graphql_desc_output_strip_backtick', 20, 3 );`
	`function graphql_desc_output_strip_backtick( $translated_text, $text, $domain ) {
		if ( $domain === 'wp-graphql' && ( strpos($text, '`') !== false ) ) {
			$new_text = preg_replace('/`(.*?)`/', ' $1 ', $text);
			$translated_text = __( $new_text, $domain );
		}
		return $translated_text;
	}
```
But not sure I want to be doing that.

**Comment by jasonbahl** [comment]
@hwlmatt thanks for the super detailed issue!

I can't recall any specific reason to include backticks inside descriptions. 

Can you point out the 35 instances you came across? Happy to check them out and see if we can simply replace them. My guess is that a single or double quote would be fine indeed.

**Comment by jasonbahl** [comment]
@hwlmatt this is the only issue that came to mind related to this: #364 

But it looks like escaped double quotes should work fine. 

I'm working on a pretty big release, so I can include this change in that release. Hoping to have a PR today for it. 😄

**Comment by hwlmatt** [comment]
Cool. And yeah, I can list off the instances I found. Let me track them all down real quick and I'll post file/line as I can.

**Comment by CodeProKid** [comment]
I think these characters are here because we simply copy & pasted from the WordPress codex / Developer reference for a lot of these. It can probably be cleaned up.

**Comment by jasonbahl** [comment]
@hwlmatt I believe @CodeProKid is correct. 

I just updated all cases that I found in the PR I'm working on, so this will be fixed as part of the PR for #327

**Comment by jasonbahl** [comment]
Should probably do a general Description Audit at some point. Reading through some of these descriptions makes me think a lot of them could be written better . . . another symptom of copy/paste from the Codex which was not written with WPGraphQL in mind.

**Comment by hwlmatt** [comment]
@jasonbahl 
Cool. If you want a double check, I believe the unique values are:
```
`$name__like`
`$description__like`
`comment_author_IP`
`date`
`date_gmt`
`comment_content`
`comment_karma`
`comment_approved`
`comment_agent`
`comment_type`
`post_objects`
`wp/v2`
`searchColumns`
`wp_capabilities`
`users`
```

And the locations and context by file:

**TermObjectConnectionArgs.php**

```
line 106
'description' => __( 'Retrieve terms with criteria by which a term is LIKE `$name__like`. Default empty.', 'wp-graphql' ),

line: 110
'description' => __( 'Retrieve terms where the description is LIKE `$description__like`. Default empty.', 'wp-graphql' ),

```
###

**/src/Type/Comment/CommentType.php**

```
line 109 
'description' => __( 'IP address for the author. This field is equivalent to WP_Comment->comment_author_IP and the value matching the `comment_author_IP` column in SQL.', 'wp-graphql' ),

line 116
'description' => __( 'Date the comment was posted in local time. This field is equivalent to WP_Comment->date and the value matching the `date` column in SQL.', 'wp-graphql' ),

line 123
'description' => __( 'Date the comment was posted in GMT. This field is equivalent to WP_Comment->date_gmt and the value matching the `date_gmt` column in SQL.', 'wp-graphql' ),

line 130
'description' => __( 'Content of the comment. This field is equivalent to WP_Comment->comment_content and the value matching the `comment_content` column in SQL.', 'wp-graphql' ),

line 150
'description' => __( 'Karma value for the comment. This field is equivalent to WP_Comment->comment_karma and the value matching the `comment_karma` column in SQL.', 'wp-graphql' ),

line 157
'description' => __( 'The approval status of the comment. This field is equivalent to WP_Comment->comment_approved and the value matching the `comment_approved` column in SQL.', 'wp-graphql' ),

line 164
'description' => __( 'User agent used to post the comment. This field is equivalent to WP_Comment->comment_agent and the value matching the `comment_agent` column in SQL.', 'wp-graphql' ),

line 171
'description' => __( 'Type of comment. This field is equivalent to WP_Comment->comment_type and the value matching the `comment_type` column in SQL.', 'wp-graphql' ),
```

###

**src/Type/PostObject/Mutation/**

```
line 66
'description' => __( 'If the post is an attachment or a media file, this field will carry the corresponding MIME type. This field is equivalent to the value of WP_Post->post_mime_type and the post_mime_type column in the `post_objects` database table.', 'wp-graphql' ),

```
###

**src/Type/PostObject/PostObjectType.php**

```
line 259
'description' => __( 'The uri slug for the post. This is equivalent to the WP_Post->post_name field and the post_name column in the database for the `post_objects` table.', 'wp-graphql' ),

line 331
'description' => __( 'The global unique identifier for this post. This currently matches the value stored in WP_Post->guid and the guid column in the `post_objects` database table.', 'wp-graphql' ),
```

###

**src/Type/PostType/PostTypeType.php**

```
line 192
'description' => __( 'Whether to add the post type route in the REST API `wp/v2` namespace.', 'wp-graphql' ),

line 199
 'description' => __( 'Name of content type to diplay in REST API `wp/v2` namespace.', 'wp-graphql' ),
```

###

 **src/Type/Taxonomy/TaxonomyType.php**

 ```
line 146
 'description' => __( 'Whether to add the post type route in the REST API `wp/v2` namespace.', 'wp-graphql' ),

 line 153
 'description' => __( 'Name of content type to diplay in REST API `wp/v2` namespace.', 'wp-graphql' ),
```

###

 **src/Type/User/Connection/UserConnectionArgs.php**

 ```
line 94
 'description' => __( 'Search keyword. Searches for possible string matches on columns. When `searchColumns` is left empty, it tries to determine which column to search in based on search string.', 'wp-graphql' ),
```

### 

**src/Type/User/UserType.php**

```
line 79
'description' => __( 'This field is the id of the user. The id of the user matches WP_User->ID field and the value in the ID column for the `users` table in SQL.', 'wp-graphql' ),

line 93
 'description' => __( 'User metadata option name. Usually it will be `wp_capabilities`.', 'wp-graphql' ),
```

###

**Comment by hwlmatt** [comment]
I got the number 35 from the output schema, so there are a lot of duplicated occurrences, of course.

**Comment by jasonbahl** [comment]
@hwlmatt thanks! This looks like it matches what I found! 

Assuming the PR I'm about to drop doesn't break everything (it might, lol) this will be fixed in the next release 😄

**Comment by hwlmatt** [comment]
@jasonbahl Also, regarding the similarity with the issue #364 you mentioned above, curious if Apollo Server 2.0's use of graphql-tools and the backticks (and non-escaped quotes) are the source of an issue I had previously with schema introspection failing on remote schema. 
In any event, I would consider this merely an elaboration on the issue in #364 .

**Comment by jasonbahl** [comment]
closed via #558
